core/shared/src/main/scala/org/http4s/Message.scala

```diff
@@ -503,6 +490,11 @@ final class Request[F[_]] private (
 
 object Request {
 
+  /** A [[Request]] that doesn't have a body requiring effectful evaluation.
+    *  A [[Request.Pure]] is always an instance of [[Request[F]]], regardless of `F`.
+    */
+  type Pure = Request[fs2.Pure]
```

Comment:
No, that won't work because fs2.Pure is defined as a type inequality: `type Pure[A] <: Nothing`. Even if we define our own `Pure` type in the exact same way, the compiler won't unify them. The only way I can see to avoid referring to fs2 here is to push the definition up into Cats.

Reply:
> The only way I can see to avoid referring to fs2 here is to push the definition up into Cats.

I think that's worth a PR.

Reply:
But also, could we define our own `type Pure[A] = fs2.Pure[A]`?

Reply:
> I think that's worth a PR.
Sure, I'll give it a go

> But also, could we define our own `type Pure[A] = fs2.Pure[A]`?
There's nothing to stop us - I'm not wild about the extra indirection though.

Reply:
I opened a PR at https://github.com/typelevel/cats/pull/4158 but I don't know if I did a very good job of explaining the rationale.

Reply:
I mean, does it matter that we expose an fs2 type? `Stream` is exposed all over the place

Reply:
I don't personally have strong feelings about it. I do think that as a matter of principle, cats is a better place for it to be defined - it should in principle be possible for another library that doesn't have an fs2 dependency to be able to use `Pure`/`INothing` and be fully interoperable with the fs2 types, which isn't currently possible.

Reply:
This seems like a lot of change for little relative benefit.

Reply:
Why a lot of change? Cats adds it, fs2 updates its aliases, nobody is none the wiser.

Reply:
What's the benefit though?

Reply:
We should move discussion to https://github.com/typelevel/cats/pull/4158.

Reply:
After a bit more fiddling, I think we may be less tied to `fs2.Pure` than I thought, rendering this moot.